x-pack/plugins/apm/public/components/alerting/rule_types/transaction_error_rate_rule_type/index.tsx

```diff
@@ -99,15 +99,23 @@ export function TransactionErrorRateRuleType(props: Props) {
   const fields = [
     <ServiceField
       currentValue={params.serviceName}
-      onChange={(value) => setRuleParams('serviceName', value)}
+      onChange={(value) => {
+        if (value !== params.serviceName) {
+          setRuleParams('serviceName', value);
+          setRuleParams('transactionType', '');
+          setRuleParams('environment', ENVIRONMENT_ALL.value);
+        }
+      }}
```

Comment:
What if we rewrite the `setRuleParams` like this

```
  function setRuleParams(properties: Partial<RuleParams>) {
    setParams((oldParams) => ({ ...oldParams, ...properties }));
  }
```

then we can batch all these calls in a single call

```
setRuleParams({
    'service': value,
    'transactionType': '',
    'environment': ENVIRONMENT_ALL.value
});
```

Reply:
Plus this implementation will provide type safety also
<img width="831" alt="image" src="https://user-images.githubusercontent.com/7416358/215540407-77577e52-cbbe-4ed7-a11c-22c43d7b92ec.png">

Reply:
looks good to me, thanks for the suggestion.

Reply:
ah, the thing is the current `setRuleParams` receives a key and a string `setRuleParams: (key: string, value: any)`, so I'm not able to update the entire object at once.

Reply:
Yup, thats why i recommended if its possible to rewrite this function. But if it's not then ok

>   function setRuleParams(properties: Partial<RuleParams>) {
    setParams((oldParams) => ({ ...oldParams, ...properties }));
  }

Reply:
In this particular case, this file has the implementation of `setRuleParams` which can be rewritten
https://github.com/elastic/kibana/blob/24765997bb1cf99eca5e53b3bda5404c018a528e/x-pack/plugins/apm/public/components/alerting/rule_types/error_count_rule_type/index.stories.tsx

Reply:
`setRuleParams` is something the alerts team provides us. Changing it would be a bit more work IMHO.

Reply:
> In this particular case, this file has the implementation of `setRuleParams` which can be rewritten
> [here](https://github.com/elastic/kibana/blob/24765997bb1cf99eca5e53b3bda5404c018a528e/x-pack/plugins/apm/public/components/alerting/rule_types/error_count_rule_type/index.stories.tsx#L56)

Ah, you linked to a Storybook file.